lib/main.js

```diff
@@ -149,61 +165,92 @@ if either your review configuration or your project's dependencies change.`);
   exit(0);
 }
 
-module.exports = () => {
+/**
+ * @returns {Promise<void | never>}
+ */
+async function main() {
   if (options.version) {
     console.log(Anonymize.version(options));
     return;
   }
 
-  if (options.subcommand === 'init') {
-    if (options.help) {
-      return Help.init();
+  switch (options.subcommand) {
+    case 'init': {
+      if (options.help) {
+        return Help.init();
+      }
+
+      try {
+        return await Init.promptAndCreate(options);
+      } catch (error) {
+        errorHandler(error);
+      }
     }
 
-    return Init.promptAndCreate(options).catch(errorHandler);
-  }
+    case 'new-rule': {
+      if (options.help) {
+        return Help.newRule();
+      }
 
-  if (options.subcommand === 'new-rule') {
-    if (options.help) {
-      return Help.newRule();
+      try {
+        return await NewRule.create(options);
+      } catch (error) {
+        errorHandler(error);
+      }
     }
 
-    return NewRule.create(options).catch(errorHandler);
-  }
+    case 'new-package': {
+      if (options.help) {
+        return Help.newPackage();
+      }
 
-  if (options.subcommand === 'new-package') {
-    if (options.help) {
-      return Help.newPackage();
+      try {
+        return await NewPackage.create(options);
+      } catch (error) {
+        errorHandler(error);
```

Comment:
Remind me: Does this `await + try catch` improve stack traces? Because otherwise I have to admit I don't see this as an improvement. It's more verbose and less Elm-like (in the sense of chaining `Task`s), which has its benefits too.

Also, shouldn't there be a `return` on or after `errorHandler`? Otherwise, I think we'll run the code after the `switch` statement when an error occurs.

Reply:
It should, is my understanding. I could be misremembering this though.

ErrorHandler returns never, so it's not necessary.
If you'd prefer it reverted, that's fine as well. It is more verbose.

Reply:
I think I'd prefer it reverted :shrug: